src/store/SubtleDataEncryptor.ts

```diff
@@ -58,7 +58,7 @@ export class SubtleDataEncryptor implements DataEncryptor {
 	}
 
 	stringToArrayBuffer( data: string ): ArrayBuffer {
-		return new TextEncoder().encode( data );
+		return new TextEncoder().encode( data ).buffer as ArrayBuffer;
```

Comment:
I had another look at the Typescript release notes and they mention the error message: https://devblogs.microsoft.com/typescript/announcing-typescript-5-7/#typedarrays-are-now-generic-over-arraybufferlike

Since casting in TypeScript is error-prone and can lead to subtle errors, I recommend changing the return type for `stringToArrayBuffer`. Just let it return `Uint8Array`. The receiving function, [`SubtleCrypto.encrypt`](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt) can take several types, one of them `TypedArray`. `Uint8Array` extends the `TypedArray` base class, so we're good here.